.github/workflows/swift-toolchain.yml

```diff
@@ -906,19 +904,29 @@ jobs:
 
       - uses: compnerd/gha-setup-vsdevenv@f1ba60d553a3216ce1b89abe0201213536bc7557 # main
         with:
-          host_arch: amd64
+          host_arch: ${{ inputs.build_arch }}
           components: 'Microsoft.VisualStudio.Component.VC.Tools.x86.x64;Microsoft.VisualStudio.Component.VC.Tools.ARM64'
           arch: ${{ matrix.arch }}
 
       - name: Install Swift Toolchain
+        if: inputs.build_arch == 'amd64'
         uses: compnerd/gha-setup-swift@b6c5fc1ed5b5439ada8e7661985acb09ad8c3ba2 # main
         with:
           github-repo: thebrowsercompany/swift-build
           github-token: ${{ secrets.GITHUB_TOKEN }}
-          release-asset-name: installer-amd64.exe
+          release-asset-name: installer-${{ inputs.build_arch }}.exe
           release-tag-name: '20231016.5'
+      - name: Install Swift Toolchain
+        if: inputs.build_arch == 'arm64'
+        uses: compnerd/gha-setup-swift@b6c5fc1ed5b5439ada8e7661985acb09ad8c3ba2 # main
+        with:
+          github-repo: thebrowsercompany/swift-build
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          release-asset-name: installer-${{ inputs.build_arch }}.exe
+          release-tag-name: '20231016.0'
```

Comment:
Can we find a single version that works for both and then just use the name to switch between the arches?

Reply:
Here's the context on this: https://thebrowsercompany.slack.com/archives/C07CU2UU97F/p1731980631963529 The `20231016.5` is a specially made build by Alex with a special fix for the main swift-build to work and Alex encountered a build issue for arm64 and hence there's no `20231016.5` for arm64. We cannot simply find a single version because it has a special fix. In addition, the arm64 pinned toolchain needs to be updated anyway as neither `20231016.5` or `20231016.0` or anything close won't work for the correct arm64 build. So, this is a stop-gap placeholder until we have a clearer direction for fixing the arm64 bootstrap process. We could unify when that happens.